hibike/devices/RFID/rfid.cpp

```diff
@@ -0,0 +1,98 @@
+#include "rfid.h"
+#include <SPI.h>
+#include <MFRC522.h> 
+
+#define RST_PIN 9
+#define SS_PIN 10
+
+// Instantiate RFID Object
+MFRC522 mfrc522(SS_PIN, RST_PIN);
+// each device is responsible for keeping track of it's own params
+uint8_t id;
+uint8_t tag_detect;
+bool del = false;
+
+// normal arduino setup function, you must call hibike_setup() here
+void setup() {
+  hibike_setup();
+  // Setup sensor input
+  SPI.begin();
+  mfrc522.PCD_Init();
+}
+
+
+// normal arduino loop function, you must call hibike_loop() here
+// hibike_loop will look for any packets in the serial buffer and handle them
+// hibike_loop will call device_write and device_read
+
+void loop() {
+  // do whatever you want here
+  // note that hibike will only process one packet per call to hibike_loop()
+  // so exessive delays here will affect hibike.
+  // Look for new cards
+  hibike_loop();
+  if ( ! mfrc522.PICC_IsNewCardPresent()) {
+    if (del) {
```

Comment:
Please, at least one comment in this whole loop function.

What we discovered about the consecutive reads should (and hence why we need del) should be recorded somewhere.